### PR TITLE
[REF] Standardise methods of determining isTest

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
[REF] Standardise methods of determining isTest 

Before
----------------------------------------
A bit ad hoc

After
----------------------------------------
used a function

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton so if someone was to load up the live back office new membership form but with &action=preview then at the moment would that be treated as test or not?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 on the main flow it would be unchanged - for the obscure flow it would 'do what the main flow does' - which means these lines https://github.com/civicrm/civicrm-core/pull/19417/files#diff-e7ba7553d0e3638ecd45d305f95eabe0fec73ff33609a9cfb4cf0f148fb118e0L1954 (written for the front end pages) would not apply - which makes sense as we should not do something different for recurring than non recurring (which is the main flow) wrt is_test

File: CRM/Member/Form/Membership.php [hunk 1]
```diff
@@ -1950,12 +1948,7 @@ protected function legacyProcessRecurringContribution(array $params, $contactID)
     $recurParams['currency'] = $params['currency'] ?? NULL;
     $recurParams['payment_instrument_id'] = $params['payment_instrument_id'];
 
-    $recurParams['is_test'] = 0;
-    if (($form->_action & CRM_Core_Action::PREVIEW) ||
-      (isset($form->_mode) && ($form->_mode == 'test'))
```
Comment (monishdeb) [inline CRM/Member/Form/Membership.php:1955]: ensured removing $form->_action & CRM_Core_Action::PREVIEW doesn't affect the test mode processing on membership backoffice form.

File: CRM/Member/Form/Membership.php [hunk 2]
```diff
@@ -1938,7 +1937,6 @@ protected function processContribution(
    * @return int
    */
   protected function legacyProcessRecurringContribution(array $params, $contactID): int {
-    $form = $this;
```
Comment (monishdeb) [inline CRM/Member/Form/Membership.php:1941]: safely deleted, no reference to $form present.

**Comment by monishdeb** [comment]
Did r-run on local, doesn't break scenario for test/live mode. Reviewed code, looks good to me.

**Comment by eileenmcnaughton** [comment]
Thanks @monishdeb

**Pull request merged**